### ISSUE TITLE
Provide ability to comment out regions of content in visual mode

### ISSUE THREAD

**Issue by jjallaire** [rstudio/rstudio#7583]
Originally reported/discussed here: https://github.com/rstudio/rstudio/issues/7581

Currently the visual mode 'Add Comment' command (bound to Cmd+Shift+C) is intended to insert editing/review comments (and provides a special background highlight to denote this). However, users may also want to "comment out" regions of markdown in the same way that they do in source mode.

Note that this is trickier than it appears b/c once you comment out markdown Pandoc no longer parses it. So for example here is what a section commented out in source mode looks like in visual mode:

<img width="682" alt="Screen Shot 2020-08-14 at 6 19 45 AM" src="https://user-images.githubusercontent.com/104391/90239855-3c34ad80-ddf6-11ea-9bdb-504b59f9dd75.png">

We certainly *could* have a gesture that does this (and undoes it by re-rendering the markdown) but the implementation won't be as straightforward as just inserting a comment marker.

**Comment by jjallaire** [comment]
I don't think this is something we can or should do for 1.4 (too expensive/complicated)

**Comment by mikebessuille** [comment]
@jjallaire in which release do you want to consider this?

**Comment by jjallaire** [comment]
We will have a likely do a bunch of work in the visual editor in 2023 and this may get picked up as part of that (we'll certainly review every open visual editor bug as a starting point)
